src/openyieldtables/yieldtables.py

```diff
@@ -189,3 +190,93 @@ def get_yield_table(id: int) -> YieldTable:
         **yield_table_meta.model_dump(),
         data=YieldTableData(yield_classes=yield_classes),
     )
+
+
+def get_interpolated_yield_table(id: int, target_yield_class: float) -> YieldClass:
+    """
+    Reads the yield table data for a specific yield table ID and yield class from
+    `data/yield_tables.csv` and returns an interpolated YieldClass instance.
+
+    The CSV file is expected to be in a specific format, with columns for
+    `id`, `yield_class`, `age`, `dominant_height`, `average_height`, etc.
+
+    Args:
+        id: The ID of the yield table to get the data for.
+        target_yield_class: The target yield class to interpolate to.
+
+    Returns:
+        YieldClass: A `YieldClass` instance for the target yield class.
+
+    """
+
+    # Make a list of yield class rows
+    lower_yield_class = int(target_yield_class)
+    upper_yield_class = lower_yield_class + 1
```

Comment:
This could be a problem since the lower yield class is not always the next lower full number. A yield class can also be a float. And the gap between the yield classes is not always `1`.